pkg/infra/usagestats/usage_stats.go

```diff
@@ -99,6 +99,28 @@ func (uss *UsageStatsService) GetUsageReport(ctx context.Context) (UsageReport,
 		} else {
 			dsOtherCount += dsStat.Count
 		}
+
+		if dsStat.Type == models.DS_ES {
+			dsSettings := models.GetESJSONDataQuery{}
+			if err := uss.Bus.Dispatch(&dsSettings); err != nil {
```

Comment:
If I remember correctly I've heard of users having several thousands of elasticsearch data sources. We don't want the usage stats to negatively impact the overall performance of Grafana. Hard to tell without testing. Should there be a risk of impacting performance negatively we always have the option to introduce a setting to disable reporting of these stats even though I would recommend trying to keep the number of configuration settings as low as possible.

Reply:
thanks for pointing this out, I'll run some benchmarks and post the results here 👍

Reply:
I tried to benchmark the change with a set of 10000 configured es datasources and the results are: 

```
BenchmarkMetrics       	     136	   8505692 ns/op	 7191026 B/op	   59823 allocs/op
BenchmarkMetrics-2     	     194	   6069296 ns/op	 7191211 B/op	   59824 allocs/op
BenchmarkMetrics-4     	     218	   5402026 ns/op	 7191428 B/op	   59824 allocs/op
BenchmarkMetrics-8     	     214	   5533107 ns/op	 7191802 B/op	   59824 allocs/op
BenchmarkMetrics-16    	     211	   5706933 ns/op	 7192659 B/op	   59825 allocs/op
```

Which should be ok. For the test the DB is in memory though, which doesn't really make it very accurate. 

i also tried to collect some data on the number of configured ES datasources per grafana instance and it should be pretty safe (the avg as of yesterday was 6.02, with 95th percentile being 16)